hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdb/util/TestManagedSstFileReader.java

```diff
@@ -91,65 +95,131 @@ private Map<String, Integer> createKeys(int startRange, int endRange) {
+                        .orElse(true))
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                    Map.Entry::getValue));
+        try (Stream<String> keyStream =
+                 new ManagedSstFileReader(files).getKeyStream(
+                     lowerBound, upperBound)) {
+          keyStream.forEach(key -> {
+            Assertions.assertEquals(keysInBoundary.get(key), 1);
+            Assertions.assertNotNull(keysInBoundary.remove(key));
+          });
+          keysInBoundary.values().forEach(val -> Assertions.assertEquals(0, val));
+        }
+      }
     }
   }
 
   @Native(ROCKS_TOOLS_NATIVE_LIBRARY_NAME)
   @ParameterizedTest
   @ValueSource(ints = {0, 1, 2, 3, 7, 10})
-  @Disabled("HDDS-8818")
   public void testGetKeyStreamWithTombstone(int numberOfFiles)
       throws RocksDBException, IOException, NativeLibraryNotLoadedException {
     Assumptions.assumeTrue(NativeLibraryLoader.getInstance()
         .loadLibrary(ROCKS_TOOLS_NATIVE_LIBRARY_NAME));
-    Pair<Map<String, Integer>, List<String>> data =
+    Pair<TreeMap<String, Integer>, List<String>> data =
         createDummyData(numberOfFiles);
     List<String> files = data.getRight();
-    Map<String, Integer> keys = data.getLeft();
+    TreeMap<String, Integer> keys = data.getLeft();
     ExecutorService executorService = new ThreadPoolExecutor(0,
-        1, 60, TimeUnit.SECONDS,
+        2, 60, TimeUnit.SECONDS,
         new SynchronousQueue<>(), new ThreadFactoryBuilder()
         .setNameFormat("snapshot-diff-manager-sst-dump-tool-TID-%d")
         .build(), new ThreadPoolExecutor.DiscardPolicy());
     ManagedSSTDumpTool sstDumpTool =
         new ManagedSSTDumpTool(executorService, 256);
 
-    try (Stream<String> keyStream = new ManagedSstFileReader(files)
-        .getKeyStreamWithTombstone(sstDumpTool)) {
-      keyStream.forEach(keys::remove);
-      Assertions.assertEquals(0, keys.size());
+    List<Optional<String>> bounds = getTestingBounds(keys);
+    try {
+      for (Optional<String> lowerBound : bounds) {
+        for (Optional<String> upperBound : bounds) {
+          Map<String, Integer> keysInBoundary =
+              keys.entrySet().stream().filter(entry -> lowerBound
+                      .map(l -> entry.getKey().compareTo(l) >= 0)
+                      .orElse(true)  &&
+                      upperBound.map(u -> entry.getKey().compareTo(u) < 0)
+                          .orElse(true))
+                  .collect(Collectors.toMap(Map.Entry::getKey,
+                      Map.Entry::getValue));
+          try (Stream<String> keyStream = new ManagedSstFileReader(files)
+              .getKeyStreamWithTombstone(sstDumpTool, lowerBound, upperBound)) {
+            keyStream.forEach(
+                key -> {
+                  Assertions.assertNotNull(keysInBoundary.remove(key));
+                });
+          }
+          Assertions.assertEquals(0, keysInBoundary.size());
+        }
+      }
```

Comment:
Please add comments around such lambda to clarify the intention for others to review.

Reply:
done